qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/fitting/plotting/plot_view.py

```diff
@@ -209,10 +212,11 @@ def remove_ws_from_fitbrowser(self, ws):
         except:
             pass  # name may not be available if ws has just been deleted
 
-    def update_legend(self, ax):
+    def update_legend(self, ax, show_legend):
         if ax.get_lines():
             ax.make_legend()
             ax.get_legend().set_title("")
+            ax.get_legend().set_visible(show_legend)
```

Comment:
Here instead of using a function argument, you can do
`ax.get_legend().set_visible(self.toolbar.get_show_legend_value())`

I think this is better because automatically updates all older cases that might call the function `update_legend`

Reply:
> Here instead of using a function argument, you can do `ax.get_legend().set_visible(self.toolbar.get_show_legend_value())`
> 
> I think this is better because automatically updates all older cases that might call the function `update_legend`

This is changed as suggested, Thanks for noting that.